Core/ComponentBaseClasses/elxTransformBase.h

```diff
@@ -96,6 +96,7 @@ namespace elastix
  * \transformparameter Direction: The direction cosines matrix of the fixed image
  * that was used during registration, and which is used for resampling the deformed moving image
  * if the UseDirectionCosines parameter is set to "true".\n
+ * Please note: The matrix elements should be specified in column-major order!\n
```

Comment:
I would describe more explicitly that this differs from ITK conventions:
```suggestion
 * Please note: The matrix elements are specified in column-major order (Fortran-style).
 * This is not the same as ITK's convention of using row-major order (C-style) for flattening
 * direction matrix of an image into a list.\n
```

Reply:
I think @lassoan's comments reduces the room for error even further!

Reply:
@lassoan Thank you for your suggestion 👍 I'm fine with your suggestion, at least if it's the only place where it should the documented. (If this specification should be documented on multiple places, I'd rather elaborate on the meaning of "column-major order" just once, rather than on each of those places. But this appears the one and only place where it needs to be documented, right?)

One nit-pick question, why do you suggest replacing
> The matrix elements _should be_ specified in column-major order!

with

> The matrix elements _are_ specified in column-major order!
?

I wanted to express that the user _should_ specify the matrix elements in that order.

Reply:
I agree, this detailed note should only appear once in the Elastix documentation. So, if this needs to be mentioned elsewhere in the future than that place should refer to this note or this note could be moved out elsewhere.

To me, "should" has some degree of uncertainty. If "should" is needed to make it clear that it is up to the user has to specify the values then we could use "must" or "shell" to make it clear that column-major order is not just a recommendation but a requirement.

Reply:
OK, thanks, then shall we make it "must"?

Reply:
@lassoan @sorenchr2011 Stefan (@stefanklein) just pointer out to me that MetaIO also uses column-major order for its `TransformMatrix` representation. See https://discourse.itk.org/t/metaio-image-rotation-conventions/4525 and https://itk.org/Wiki/ITK/MetaIO/Documentation#Associated_transformations

So then, is ITK's convention really _using row-major order_?  Shouldn't the comment say something like:

```
Please note: The matrix elements are specified in column-major order (Fortran-style, like MetaIO's matrix representation).
ITK often uses row-major order (C-style) for flattening direction matrix of an image into a list.
```

Reply:
> @lassoan @sorenchr2011 Stefan (@stefanklein) just pointer out to me that MetaIO also uses column-major order for its `TransformMatrix` representation. See https://discourse.itk.org/t/metaio-image-rotation-conventions/4525 and https://itk.org/Wiki/ITK/MetaIO/Documentation#Associated_transformations
> 
> So then, is ITK's convention really _using row-major order_?
> 

ITK internally uses row-major order. Various file formats don't have to use the same conventions as ITK. ITK-based libraries (such as Elastix) don't have to use the same conventions as ITK either, but when a user gets a vector from an ITK method then he expects to be able to use it as input in another method in an ITK-based library without any conversion, so it is worth trying to harmonize conventions. Bringing in MetaIO into this discussion is unnecessary and may be distracting.

> The matrix elements are specified in column-major order (Fortran-style, like MetaIO's matrix representation).
ITK often uses row-major order (C-style) for flattening direction matrix of an image into a list.

I don't think this is accurate, because ITK is consistent in how it stores matrices in memory and how they may be exposed as flat arrays via Python wrapping.

Anyway, any documentation would improve things and I don't really mind the details.